docs/spec/spec.md

```diff
@@ -2075,11 +2088,171 @@ service on new graphql:Listener(4000) {
+
+###### Example: Client Using Declarative Basic Authentication
+
+```ballerina
+public function main() returns error? {
+    graphql:Client graphqlClient = check new ("http://localhost:4000/graphql",
+        auth = {
+            username: "bob",
+            password: "bob@123"
+        }
+    );
+    json response = check graphqlClient->execute("{ greeting }");
+    // ...
+}
+```
+
+##### 11.2.2 Bearer Token Authentication
+
+Ballerina GraphQL clients enable authentication using bearer tokens by setting the `graphql:BearerTokenConfig` configurations in the client. The requests from the client are automatically enriched with the `Authorization: Bearer <token>` header when passing the `graphql:BearerTokenConfig` for the `auth` configuration of the client.
+
+###### Example: Client Using Declarative Bearer Token Authentication
+
+```ballerina
+public function main() returns error? {
+    graphql:Client graphqlClient = check new ("http://localhost:4000/graphql",
+        auth = {
+            token: "56ede317-4511-44b4-8579-a08f094ee8c5"
+        }
+    );
+    json response = check graphqlClient->execute("{ greeting }");
+    // ...
+}
+```
+
+##### 11.2.3 Self Signed JWT Authentication
+
+Ballerina GraphQL clients enable authentication using JWTs by setting the `graphql:JwtIssuerConfig` configurations in the client.  The requests from the client are automatically enriched with the `Authorization: Bearer <token>` header when passing the `graphql:JwtIssuerConfig` for the `auth` configuration of the client.
+
+###### Example: Client Using Declarative Self Signed JWT Authentication
+
+```ballerina
+public function main() returns error? {
+    graphql:Client graphqlClient = check new ("http://localhost:4000/graphql",
+        auth = {
+            username: "ballerina",
+            issuer: "wso2",
+            audience: ["ballerina", "ballerina.org", "ballerina.io"],
+            keyId: "5a0b754-895f-4279-8843-b745e11a57e9",
+            jwtId: "JlbmMiOiJBMTI4Q0JDLUhTMjU2In",
+            customClaims: {"scp": "admin"},
+            expTime: 3600,
+            signatureConfig: {
+                config: {
+                    keyFile: "path/to/private.key"
+                }
+            }
+        }
+    );
+    json response = check graphqlClient->execute("{ greeting }");
+    // ...
+}
+```
+
+##### 11.2.4 OAuth2
+
+Ballerina GraphQL clients enable authentication using OAuth2 by setting the `graphql:OAuth2GrantConfig` configurations in the client. The requests from the client are automatically enriched with the `Authorization: Bearer <token>` header when passing the `graphql:OAuth2GrantConfig` for the `auth` configuration of the client. OAuth2 can configure in following 4 ways
+
+###### 11.2.4.1 Client Credentials Grant Type
```

Comment:
Shall we add a small description (or link to the specific sections in the OAuth2 spec) for different grant types?

Reply:
Keeping this as it's for now since oauth2 module doesn't have separate section for grant type instead it's points to an rfc